lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosSobolIndices.cxx

```diff
@@ -350,6 +350,50 @@ Scalar FunctionalChaosSobolIndices::getSobolGroupedIndex(const UnsignedInteger v
   return getSobolGroupedIndex(index, marginalIndex);
 }
 
+/* Sobol grouped total index accessor */
+Scalar FunctionalChaosSobolIndices::getSobolGroupedTotalIndex(const Indices & variableIndices,
+    const UnsignedInteger marginalIndex) const
+{
+  // Compute the complementary indice
+  const UnsignedInteger inputDimension = functionalChaosResult_.getDistribution().getDimension();
+  const UnsignedInteger groupSize = variableIndices.getSize();
+  const UnsignedInteger complementaryGroupSize = inputDimension - groupSize;
+  Point allVariableIndices(inputDimension);
+  // Initialize
+  for (UnsignedInteger j = 0; j < inputDimension; ++j)
+  {
+    allVariableIndices[j] = j;
+  }
+  // Browse all variables : set -1 to the variables not in the complementary group
+  for (UnsignedInteger i = 0; i < groupSize; ++i)
+  {
+    allVariableIndices[variableIndices[i]] = -1;
+  }
+  // Set the complementary indices
+  Indices complementaryVariableIndices(complementaryGroupSize);
+  UnsignedInteger i = 0;
+  for (UnsignedInteger j = 0; j < inputDimension; ++j)
+  {
+    if (allVariableIndices[j]>=0)
+    {
+        complementaryVariableIndices[i] = UnsignedInteger(allVariableIndices[j]);
+        i=i+1;
+    }
+  }
+  // Compute total index from complementary first index
+  Scalar complementaryFirstIndex = getSobolGroupedIndex(complementaryVariableIndices, marginalIndex);
+  Scalar groupTotalIndex = 1.0-complementaryFirstIndex;
+  return groupTotalIndex;
+}
+/* Sobol total index accessor */
+Scalar FunctionalChaosSobolIndices::getSobolGroupedTotalIndex(const UnsignedInteger variableIndex,
+    const UnsignedInteger marginalIndex) const
+{
+  Indices index(1);
+  index[0] = variableIndex;
```

Comment:
you can use the ctor: `Indices(1, variableIndex) `

Reply:
Correct. Actually, the same statements are here and there in the FunctionalChaosSobolIndices.cxx: I replace them all with your suggestion.